### PR TITLE
Initial implementation for metrics

### PR DESCRIPTION
**Pull request by kozlovzxc-at-lido**
## Description

Added common metrics:
- Default node performance metrics
- Startup configuration metrics
- External requests metrics
- Internal API responses metrics

Added RPC rate limiting

## Demo

App is working
image

Metrics are there
image

## Review notes

Do we need any other metrics?

## Testing notes

Need to check that everything works as normal, no new functionality is expected.

### CODE REVIEW

**Review by mvaleev — APPROVED** [review]

File: features/metrics/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { collectDefaultMetrics, Registry } from 'prom-client';
+import { dynamics } from 'config';
+import buildInfoJson from 'build-info.json';
+import { collectStartupMetrics } from '@lidofinance/api-metrics';
+import { METRICS_PREFIX } from './config';
+import { RequestMetrics } from './request-metrics';
+
+class Metrics {
```
Comment (Jabher) [inline features/metrics/metrics.ts:8]:
```suggestion
export const metrics = new class Metrics {
```

File: features/metrics/metrics.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  registry = new Registry();
+
+  request = new RequestMetrics(this.registry);
+
+  constructor() {
+    this.collectStartupMetricsInit();
```
Comment (Jabher) [inline features/metrics/metrics.ts:14]: why not inlined?

File: features/metrics/request-metrics.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export class RequestMetrics {
+  apiTimings: Histogram<'hostname' | 'route' | 'entity' | 'status'>;
+
+  constructor(public registry: Registry) {
+    this.apiTimings = new Histogram({
+      name: METRICS_PREFIX + 'api_response_internal',
+      help: 'API response time',
+      labelNames: ['hostname', 'route', 'status'],
+      buckets: [0.1, 0.2, 0.3, 0.6, 1, 1.5, 2, 5],
+      registers: [this.registry],
```
Comment (Jabher) [inline features/metrics/request-metrics.ts:13]:
```suggestion
      registers: [registry],
```

for better readability

File: features/metrics/response-time-metrics.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    try {
+      await next?.(req, res, next);
+      status = getStatusLabel(res.statusCode);
+    } catch (error) {
+      status = getStatusLabel(res.statusCode);
+      // throw error up the stack
+      throw error;
+    } finally {
+      endMetric({ status });
+    }
```
Comment (Jabher) [inline features/metrics/response-time-metrics.ts:20]:
```suggestion
    const endMetric = metrics.startTimer({ route });

    try {
      await next?.(req, res, next);
    } finally {
      endMetric({ status: getStatusLabel(res.statusCode) });
    }
```

File: pages/api/rpc.ts [hunk 5]
```diff
@@ -1,98 +1,44 @@
...
+
+const rpc = rpcFactory({
+  fetchRPC,
+  serverLogger,
+  metrics: {
+    prefix: METRICS_PREFIX,
+    registry: metrics.registry,
+  },
+  allowedRPCMethods: [
+    'test',
```
Comment (Jabher) [inline pages/api/rpc.ts:21]: 👀

**Review by Jabher — APPROVED** [review]
approved with comments

File: features/metrics/metrics.ts (hunk 1 above)
Comment (kozlovzxc-at-lido) [reply features/metrics/metrics.ts]: Updated 👌

File: features/metrics/metrics.ts (hunk 2 above)
Comment (kozlovzxc-at-lido) [reply features/metrics/metrics.ts]: Inlined 👌

File: features/metrics/request-metrics.ts (hunk 3 above)
Comment (kozlovzxc-at-lido) [reply features/metrics/request-metrics.ts]: 👍

File: features/metrics/response-time-metrics.ts (hunk 4 above)
Comment (kozlovzxc-at-lido) [reply features/metrics/response-time-metrics.ts]: Makes sense 👍  Updated

File: pages/api/rpc.ts (hunk 5 above)
Comment (kozlovzxc-at-lido) [reply pages/api/rpc.ts:25]: It's needed by one wallet, even though it's non-existing method 👀 added comment

**Pull request merged**